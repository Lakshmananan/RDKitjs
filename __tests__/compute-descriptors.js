'use strict';

const RDKit = require('..');

beforeEach(() => RDKit.load());

test('calc_all_desc', function() {
  var smi = 'CCCCCOC(CO)';
  var mol = RDKit.Molecule.fromSmiles(smi);
  var c = RDKit.calc_all_desc(mol);
  c = JSON.stringify(c);
  expect(c).toMatchSnapshot();
  mol.delete();
});

test('getCrippenAtomContribs', function() {
  var smi = 'CO';
  var mol = RDKit.Molecule.fromSmiles(smi);
  var cp = mol.getCrippenAtomContribs();
  var LP = [];
  for (i = 0; i < cp.size() / 2; i++) {
    LP.push(cp.get(i));
  }
  var MR = [];
  for (i = cp.size() / 2; i < cp.size(); i++) {
    MR.push(cp.get(i));
  }
  expect(LP).toEqual([-0.2035, -0.2893]);
  expect(MR).toEqual([2.753, 0.8238]);

  mol.delete();
});

test('getTPSAAtomContribs', function() {
  var smi = 'CCCCCOC';
  var mol = RDKit.Molecule.fromSmiles(smi);
  var cp = mol.getTPSAAtomContribs();
  var TPSA = [];
  for (i = 0; i < cp.size(); i++) {
    TPSA.push(cp.get(i));
  }
  expect(TPSA).toEqual([0, 0, 0, 0, 0, 9.23, 0]);
  mol.delete();
});

test('getASAContribs', function() {
  var smi = 'CCC';
  var mol = RDKit.Molecule.fromSmiles(smi);
  var cp = mol.getASAContribs();
  var ASA = [];
  for (i = 0; i < cp.size(); i++) {
    ASA.push(cp.get(i));
  }
  expect(ASA).toMatchSnapshot();
  mol.delete();
});
