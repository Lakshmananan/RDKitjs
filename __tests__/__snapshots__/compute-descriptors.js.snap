// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calc_all_desc 1`] = `"{\\"mw\\":132,\\"exactMW\\":132.115029752,\\"formula\\":\\"C7H16O2\\",\\"frsp3\\":1,\\"mqn\\":[7,0,0,0,0,0,0,0,0,2,0,9,8,0,0,0,0,0,6,4,2,1,1,0,0,2,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],\\"peovsa\\":[9.843390348640755,0,0,0,0,19.765380445542643,6.4208216229260096,6.606881964512918,13.213763929025836,0,0,0,0,0],\\"smrvsa\\":[9.843390348640755,0,0,0,26.186202068468653,19.820645893538753,0,0,0,0],\\"tpsa\\":29.46,\\"slogpvsa\\":[0,24.927173288379457,4.736862953800049,0,26.186202068468653,0,0,0,0,0,0,0],\\"logp\\":1.1855,\\"mr\\":37.42979999999999,\\"labuteASA\\":56.83668269591208,\\"lipinskiHBD\\":1,\\"lipinskiHBA\\":2,\\"numHeterocycles\\":0,\\"numRings\\":0,\\"numHeteroatoms\\":2,\\"numHBA\\":2,\\"numHBD\\":1,\\"numAliphaticCarbocycles\\":0,\\"numAliphaticHeterocycles\\":0,\\"numSaturatedCarbocycles\\":0,\\"numSaturatedHeterocycles\\":0,\\"numAliphaticRings\\":0,\\"numAromaticRings\\":0,\\"numSaturatedRings\\":0,\\"numAmideBonds\\":0,\\"numRotatableBonds\\":6,\\"numAromaticHeterocycles\\":0,\\"chi0n\\":6.098102573083107,\\"chi1n\\":3.6006848163930116,\\"chi2n\\":2.043086014632321,\\"chi3n\\":1.1278531854030205,\\"chi4n\\":0.6207359402846874,\\"chi0v\\":6.098102573083107,\\"chi1v\\":3.6006848163930116,\\"chi2v\\":2.043086014632321,\\"chi3v\\":1.1278531854030205,\\"chi4v\\":0.6207359402846874,\\"kappa1\\":8.92,\\"kappa2\\":7.919999999999998,\\"kappa3\\":7.920000000000001,\\"hallKierAlpha\\":-0.08}"`;

exports[`getASAContribs 1`] = `
Array [
  6.923737199690624,
  6.4208216229260096,
  6.923737199690624,
]
`;
